conda/activate.py

```diff
@@ -459,6 +459,8 @@ def _update_prompt(self, set_vars, conda_prompt_modifier):
     def _default_env(self, prefix):
         if normpath(prefix) == normpath(context.root_prefix):
             return 'base'
+        if dirname(normpath(prefix)) in [normpath(x) for x in context.envs_dirs]:
+            return basename(prefix)
         return basename(prefix) if basename(dirname(prefix)) == 'envs' else prefix
```

Comment:
You could turn the list comprehension into a generator expression and I think we can remove the subsequent `basename(dirname(prefix)) == 'envs'` since the intention of this should be covered by your changes, i.e.:

```diff
-        if dirname(normpath(prefix)) in [normpath(x) for x in context.envs_dirs]:
+        if dirname(normpath(prefix)) in (normpath(x) for x in context.envs_dirs):
             return basename(prefix)
-        return basename(prefix) if basename(dirname(prefix)) == 'envs' else prefix
+        return prefix
```

Reply:
The implementation here is clearly the correct logic.  The hard-coded `'envs'` was probably a shortcut I took at some point figuring I'd swing back to get it correct, and then forgot because I never made a note to follow up.  So that means modifying the failing `/not-in-envs-dirs/envs/named-end` test as appropriate.

In the current master branch, we have a `paths_equal` function, but I think it's buried as an inner function inside of one of the methods.  We should pull that out and use it here (and also on line 160).

Also, please use the generator expression as pointed out by @mbargull.